### PR TITLE
Add support for ruby 3 versions

### PR DESCRIPTION
**Pull request by pragya-sriharsh**
## Why?

I am working on ruby & rails upgrade stories. As this branch is restricted to ruby 2.4 version so i have modified to support > ruby 3 versions. Initially i have modified for ruby 3.2.2 version but lessonly's some gems have some dependencies on ruby 3.0.0 so i degraded from 3.2.2 to 3.0.0. But this changes is working as expected for ruby 3.0.x too. 

## What?

Modified files to support ruby 3 version:
- changes in .ruby-version
- changes in .gemspec  that restricts previously to 2.4 ruby version
- changes in .tool-versions
- travis.yml

## Caveats
I don't know where all this gem is being used. Currently i am using it along with lessonly with this feature branch. 

## Testing Notes
NA

## Merge Instructions

Please **DO NOT** squash my commits when merging

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.7.2
+3.2.2
```
Comment (JJGayso) [inline .ruby-version:1]: Was this meant to be brought to 3.0.0 based on your PR comment?

**Review by JJGayso — CHANGES_REQUESTED** [review]

**Review by JJGayso — DISMISSED** [review]

**Comment by pragya-sriharsh** [comment]
A small change in meta to indicate ruby versions >=2.4 and <=3.2

**Review by nayan-guptaS — DISMISSED** [review]

**Comment by unixmonkey** [comment]
Why cap the upper bound at all? I would bet it works fine with Ruby 3.3 and up.

**Comment by pragya-sriharsh** [comment]
I just thought to mention that otherwise >=2.4 was enough.  Should i remove ?

**Comment by unixmonkey** [comment]
If you leave this library locked to `<=3.2`, then around the end of the year, when Ruby 3.3 final is released, then this gem will need another release, and potentially block people from upgrading to 3.3 if they use this as a dependency. Maybe for years until someone with commit access finally gets around to it, like has happened already.

In the small case that maybe something with Ruby 3.3 or higher does somehow "break" with this gem, I'm sure you'll hear about it, and then can work on an actual code fix, instead of re-pinning an upper bound, and re-releasing the gem every time a new Ruby is released.

Yes, I would recommend removing the upper bound entirely, at least until you know for sure a version won't work with it.

**Comment by pragya-sriharsh** [comment]
Okay got it @unixmonkey .  Reverting this commit.

**Review by nayan-guptaS — APPROVED** [review]

**Review by prasadb05 — APPROVED** [review]
QA testing is not required to test this PR so approving it as discussed with @JJGayso 
MicrosoftTeams-image (14)

**Pull request merged**